mailstore-connector/src/com/axonivy/connector/oauth/ssl/SSLContextConfigure.java

```diff
@@ -0,0 +1,87 @@
+package com.axonivy.connector.oauth.ssl;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.util.Properties;
+import java.util.stream.Stream;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+
+import org.apache.commons.lang3.BooleanUtils;
+
+import com.axonivy.connector.mailstore.constant.Constants;
+import com.axonivy.connector.mailstore.enums.StartTLS;
+
+public class SSLContextConfigure {
+	private static final SSLContextConfigure INSTANCE = new SSLContextConfigure();
+	private SSLContextConfigure() {}
+	
+	public static SSLContextConfigure get() {
+		return INSTANCE;
+	}
+
+	public boolean isStartTLSEnabled(Properties properties) {
+		if (properties == null) {
+			return false;
+		}
+		return BooleanUtils.toBoolean(properties.getProperty(StartTLS.ENABLE.getProperty()))
+				|| BooleanUtils.toBoolean(properties.getProperty(StartTLS.REQUIRED.getProperty()));
+	}
+
+	public void addIvyTrustStoreToCurrentContext() throws NoSuchAlgorithmException, KeyStoreException,
+			CertificateException, IOException, KeyManagementException {
+		TrustManagerFactory tmFactory = initDefaultTrustManagerFactory();
+		// Backup default Certificates
+		X509TrustManager defaultX509CertTM = getFirstX509TrustManagerFromFactory(tmFactory);
```

Comment:
Can the value of `defaultX509CertTM` be null?
If yes, maybe the system will throw NullPointerException somewhere.